### PR TITLE
487 - fix: undefined company name on email subjects

### PR DESCRIPTION
**Pull request by emmaacha**
Ticket ID: [kh-books: #487: Email Notifications - undefined company name
](https://github.com/nerubia/kh-books/issues/487)

### CODE REVIEW

File: src/services/khbooks/invoice-service.ts [hunk 1]
```diff
@@ -699,10 +699,17 @@ export const sendInvoice = async (id: number, type: string = SendInvoiceType.Inv
...
 
-  let subject = `Invoice ${invoice.invoice_no} from ${invoice.companies?.name}`
+  const companyName =
+    invoice.companies?.name !== undefined &&
+    invoice.companies?.name !== null &&
+    invoice.companies?.name.length > 0
+      ? ` from ${invoice.companies.name}`
+      : ""
+
+  let subject = `Invoice ${invoice.invoice_no}${companyName}`
```
Comment (jatnerubia) [inline src/services/khbooks/invoice-service.ts:709]: you can use the `company` above

File: src/services/khbooks/payment-service.ts [hunk 2]
```diff
@@ -506,11 +506,18 @@ export const sendPayment = async (id: number) => {
...
+    payment.companies?.name.length > 0
+      ? ` from ${payment.companies.name}`
+      : ""
+
   await sendMail({
     to: toEmails,
     cc: ccEmails,
     bcc: bccEmails,
-    subject: `Payment Receipt from ${payment.companies?.name}`,
+    subject: `Payment Receipt${companyName}`,
```
Comment (jatnerubia) [inline src/services/khbooks/payment-service.ts:520]: please use the same logic in the invoice ^^

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

File: src/services/khbooks/invoice-service.ts (hunk 1 above)
Comment (emmaacha) [reply src/services/khbooks/invoice-service.ts:707]: ^^ Done

File: src/services/khbooks/payment-service.ts (hunk 2 above)
Comment (emmaacha) [reply src/services/khbooks/payment-service.ts:525]: ^^ Done

**Review by jatnerubia — APPROVED** [review]
Cool

**Pull request merged**